interval_tree.py

```diff
@@ -0,0 +1,97 @@
+def left_child(i, grandness=1):
+    return i * 2**grandness
+
+def right_child(i, grandness=1):
+    return i * 2**grandness + 2**grandness - 1
+
+def parent(i):
+    return i // 2
+
+def _reduce(v1, v2, reducing_fn, neutral):
+    if v1 == neutral: return v2
+    if v2 == neutral: return v1
+    return reducing_fn(v1, v2)
+
+# including a, including b
+def resolve_query(tree, reducing_fn, a, b, neutral=None):
+    values = []
+    n = len(tree)
+    result = neutral
+
+    def update(i, k):
```

Comment:
update is quite a strange name for such function :) One would say it updates the tree somehow.. what about reduce_result, or combine?